contracts/contracts/core/vault/src/main.sw

```diff
@@ -919,77 +586,29 @@ impl Vault for Contract {
        / / /   |  __/| |_| | |_) | | | (__ 
       /_/_/    |_|    \__,_|_.__/|_|_|\___|
     */
-    #[storage(read, write)]
-    fn update_cumulative_funding_rate(collateral_asset: AssetId) {
-        _update_cumulative_funding_rate(collateral_asset);
-    }
-
-    /// deposit into the pool without minting RUSD tokens
-    /// useful in allowing the pool to become over-collaterised
-    #[payable]
-    #[storage(read, write)]
-    fn direct_pool_deposit(asset: AssetId) {
-        sl_require_not_paused();
-
-        _begin_non_reentrant(storage.lock);
-
-        require(
-            storage.whitelisted_assets.get(asset).try_read().unwrap_or(false),
-            Error::VaultAssetNotWhitelisted
-        );
-
-        let amount = _transfer_in(asset).as_u256();
-        // @TODO: check this
-        require(amount > 0, Error::VaultInvalidAssetAmount);
-        _increase_pool_amount(asset, amount);
-
-        log(DirectPoolDeposit {
-            asset: asset,
-            amount: amount,
-        });
-
-        _end_non_reentrant(storage.lock);
-    }
 
     #[payable]
     #[storage(read, write)]
-    fn buy_rusd(asset: AssetId, receiver: Identity) -> u256 {
+    fn add_liquidity(receiver: Identity) -> u64 {
         sl_require_not_paused();
         
-        _begin_non_reentrant(storage.lock);
-        
-        let amount_out = _buy_rusd(asset, receiver);
-        _end_non_reentrant(storage.lock);
-
-        amount_out
-    }
-
-    #[payable]
-    #[storage(read, write)]
-    fn sell_rusd(asset: AssetId, receiver: Identity) -> u256 {
-        sl_require_not_paused();
+        _begin_non_reentrant(storage::vault.lock);
```

Comment:
let's use reentrancy lib from sway libs instead our custom implementation. reentrancy from sway libs is tested + it's a standard and reduces code burden here.